platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/cache/KotlinDslWorkspaceProvider.kt

```diff
@@ -58,8 +49,9 @@ class KotlinDslWorkspaceProvider(
         kotlinDslWorkspace.close()
 
     private
-    fun subWorkspace(prefix: String): WorkspaceProvider = object : WorkspaceProvider {
-        override fun <T : Any> withWorkspace(path: String, action: WorkspaceProvider.WorkspaceAction<T>): T =
-            kotlinDslWorkspace.withWorkspace("$prefix/$path", action)
+    fun subWorkspace(prefix: String): ImmutableWorkspaceProvider = object :
+        ImmutableWorkspaceProvider {
+        override fun getWorkspace(path: String) =
+            kotlinDslWorkspace.getWorkspace("$prefix/$path")
```

Comment:
💅 
```suggestion
    fun subWorkspace(prefix: String): ImmutableWorkspaceProvider =
        object : ImmutableWorkspaceProvider {
            override fun getWorkspace(path: String) =
                kotlinDslWorkspace.getWorkspace("$prefix/$path")
```

Reply:
Thanks. IntelliJ offered to convert to a lambda, so I did.